src/module/krf.c

```diff
@@ -330,3 +340,64 @@ static ssize_t log_faults_file_write(struct file *f, const char __user *ubuf, si
   *off = buflen;
   return buflen;
 }
+
+static ssize_t targeting_file_read(struct file *f, char __user *ubuf, size_t size, loff_t *off) {
+  char buf[KRF_PROCFS_MAX_SIZE + 1] = {0};
+  size_t buflen = 0;
+  size_t offset = 0;
+  unsigned int current_mode;
+  for (current_mode = 0; current_mode < KRF_T_NUM_MODES; current_mode++) {
+    if ((krf_target_options.mode_mask & (1 << current_mode)) && (offset < KRF_PROCFS_MAX_SIZE)) {
```

Comment:
I'm worried about a buffer overflow here, but I'm not sure what the proper way to handle it in the kernel is. Should we just enlarge `KRF_PROCFS_MAX_SIZE`?

Reply:
Yeah, let's go ahead and double or quadruple it.